js/interactive-guides/utils.js

```diff
@@ -40,69 +40,115 @@ var utils = (function() {
         return resultStr;
     };
 
-    var __getNameAction = function(strAction) {
-        //console.log("AAA string ", strAction);
-        var name;  
-        if (strAction.indexOf("name=") !== -1) {
-          var index = strAction.indexOf("name=") + 5;
-          //console.log("index ", index);
-          name = strAction.substring(index);
-          //console.log("name ", name);
-          var quote = name.substring(0, 1);
-          //console.log("quote ", quote);
-          var tmpString = name.substring(1);
-          //console.log("tmpString ", tmpString);
-          var lastIndex = tmpString.indexOf(quote);
-          name = name.substring(0, lastIndex + 2);
-          //console.log("name=", name);
-        }
-        return name;
-    };
-    
-    var __getCallbackAction = function(strAction) {
-        //console.log("BBB string ", strAction);
-        var callbackStr;
-        if (strAction.indexOf("callback=") !== -1) {
-            var index = strAction.indexOf("callback=") + 9;
+    var __getQuote = function(str) {
+        var quote;
+        str = str.trimLeft();
+        quote = str.substring(0, 1);
+        //console.log("quote ", quote);
+        return quote;
+    }
+
+    var __getAttributeAction = function(strAction, attr) {
+        var str, resultStr;
+        if (strAction.indexOf(attr) !== -1) {
+            var index = strAction.indexOf(attr) + attr.length;
             //console.log("index ", index);
-            callbackStr = strAction.substring(index);
-            //console.log("callback ", callbackStr);
-            var lastIndex = callbackStr.indexOf(")");
-            callbackStr = callbackStr.substring(0, lastIndex + 2);
-            //console.log("callback=", callbackStr);
+            str = strAction.substring(index);
+            var quote = __getQuote(str);
+            if (quote === "\"" || quote === "'") {            
+                // exclude the first quote
+                var openQuoteIndex = str.indexOf(quote);
+                str = str.substring(openQuoteIndex + 1);
+                var closeQuoteIndex = str.indexOf(quote);
+                if (closeQuoteIndex !== -1) {
+                    str = quote + str.substring(0, closeQuoteIndex + 1);
```

Comment:
You should use single quote here rather than keeping the original quote. This is cuz when you put together the action tag __parseActionTag, you're assuming single quote is used. Otherwise the action tag won't work if title/onkeypressMethod/onclickMethod use double quote as the delimiter.